pkg/apis/pipeline/v1beta1/git_source.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// GitSource is an endpoint from which to get data which is required
+// by a Build/Task for context (e.g. a repo from which to build an image).
+type GitSource struct {
+	Name           string `json:"name"`
+	Type           string `json:"type"`
+	URL            string `json:"url"`
+	Branch         string `json:"branch"`
```

Comment:
`Revision` ?

In Build's `GitSource` today, we decided to accept branches, tags and arbitrary refs (including commits) in the same field, using Git's built-in semantics.

Reply:
make sense to be consistent

Reply:
Few tips:
We probably define interface `Source`  method `Type`. 
type SourceType string
enum 
```
type Source interface {
    Type() SourceType
    Version() string 
   // Pulls source code.
    Pull() error 
}
```
And Then the `GitSource` can define its own fields like you have and implement methods
```
func (gs GitSource) Version() string {
    return gs.Commit
}

func (gs GitSource) Type() SourceType {
  return constants.GitSource
} 
```

Reply:
That ways, other sources can be added easily